code/script.js

```diff
@@ -0,0 +1,269 @@
+    image: "./books-images/unknown.jpg",
+  },
+  {
+    title: "The Road",
+    author: "Cormac McCarthy",
+    year: 2006,
+    genre: "Dystopian",
+    rating: 4,
+    description:
+      "Set in a post-apocalyptic world, it follows a father and son's harrowing journey to survive and find safety.",
+    image: "./books-images/unknown.jpg",
+  },
+  {
+    title: "The Hitchhiker's Guide to the Galaxy",
+    author: "Douglas Adams",
+    year: 1979,
+    genre: "Science Fiction",
+    rating: 4.35,
+    description:
+      "A comedic science fiction series that follows the misadventures of Arthur Dent after Earth's destruction.",
+    image: "./books-images/unknown.jpg",
+  },
+  {
+    title: "The Giver",
+    author: "Lois Lowry",
+    year: 1993,
+    genre: "Dystopian",
+    rating: 4.12,
+    description:
+      "A dystopian novel set in a seemingly perfect society where young Jonas discovers the dark truth beneath the surface.",
+    image: "./books-images/unknown.jpg",
+  },
+];
+
+// DOM elements
+
+const cardCollection = document.getElementById("card-grid");
+const filterAll = document.getElementById("all");
+const filterFantasy = document.getElementById("fantasy");
+const filterFiction = document.getElementById("fiction");
+
+const showRandom = document.getElementById("show-random");
+const ratingHigh = document.getElementById("sorting")
+
+
+// function to show all arrays/objects in html
+
+const loadBooks = (bookArray) => {
+ cardCollection.innerHTML = "";
+
+  bookArray.forEach((book) => {
+    cardCollection.innerHTML += `
+    <div class="card">
+    <img src="${book.image}" class="book-image"> 
+    <h4>Title: ${book.title} </h4> 
+    <p>Author: ${book.author}</p>
+    <p>Genre: ${book.genre}</p>
+    <p>Rating: ${book.rating}</p>
+    <p>Description: ${book.description}</p>
+    </div>
+      `;
+  });
+};
+
+// call on loadBooks function to show all the book arrays inside cards on page
+
+loadBooks(books);
+
+
+
+// showing all books button
+
+const allGenres = () => {
+  const filterAll = books.filter((bookGenres) => {
+    return bookGenres.genre === "All";
+  });
+  loadBooks(books);
+};
+filterAll.addEventListener("click", allGenres);
+
+// filter books by genres buttons
+
+const fantasyGenre = () => {
+  const filterFantasy = books.filter((bookGenres) => {
+    return bookGenres.genre === "Fantasy";
+  });
+  loadBooks(filterFantasy);
+};
+filterFantasy.addEventListener("click", fantasyGenre);
+
+
+const fictionGenre = () => {
+  const filterFiction = books.filter((bookGenres) => {
+    return bookGenres.genre === "Fiction";
+  });
+  loadBooks(filterFiction);
+};
+filterFiction.addEventListener("click", fictionGenre);
+
+//sort by rating log-high
+
+const sortByRating = () => {
+  const highestRating = books.sort ((a,b) => b.rating - a.rating)
+  loadBooks(highestRating)
+};
+ratingHigh.addEventListener('click', sortByRating)
```

Comment:
you should have, following the same logic, `ratingLow` button. The sorting should always work in both ways 👍